docs/stepactions.md

```diff
@@ -113,6 +114,30 @@ spec:
     date | tee $(results.current-date-human-readable.path)
 ```
 
+### Declaring VolumeMounts
+
+You can define `VolumeMounts` in `StepAction`, note that the `name` of the `VolumeMount` MUST be a **single param reference** to a string param ( for example, `$(params.registryConfig)` is valid but `$(params.registryConfig)-foo` or `"unparametrized-name"` is invalid).
```

Comment:
must the name be a parameter? can it not just be a string?

Reply:
Yes, we explain it in the TEP. The name must be a parameter so that the Task using this step action with other steps can provide a volume device with a name they choose.

Otherwise the following can happen:

1. Lets say StepAction1 names a volumeMount `foo`
2. StepAction2 names it `bar`

From the Task perspective, it should be the same volumeDevice/workspace that binds to the mount. The Task is forced to provide two separate volume devices for `foo` and `bar` which might lead to issues. 

Parametrizing it allows the Task author to provide the name and take control of the VolumeDevice/Workspace that will bind to this mount.

Reply:
Should we include the design decision background in the user facing docs?

Reply:
Thanks for the explanation @chitrangpatel!

 
+1 to @Yongxuanzhang's comment, please add a summary of the above rationale in the docs

Reply:
added